api/src/main/java/net/kyori/adventure/pointer/PointersSupplier.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * This file is part of adventure, licensed under the MIT License.
+ *
+ * Copyright (c) 2017-2024 KyoriPowered
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package net.kyori.adventure.pointer;
+
+import java.util.function.Function;
+import net.kyori.adventure.builder.AbstractBuilder;
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * A supplier of {@link Pointers} that allows for the implementation of pointers
+ * in a static context without having to manually create a new pointers instance for
+ * each instance of a given type.
+ *
+ * <p>An example of how this could be implemented is as follows:</p>
+ * <pre>{@code
+ * public class MyPointeredObject extends SomePointeredParent implements Pointered {
+ *   private static final PointersSupplier<MyPointeredObject> POINTERS = PointersSupplier.builder()
+ *     .parent(SomePointeredParent.POINTERS) // Fallback to the parent to get pointers from.
+ *     .resolving(Identity.UUID, MyPointeredObject::getUniqueId)
+ *     .resolving(Identity.DISPLAY_NAME, MyPointeredObject::getDisplayName)
+ *     .build();
+ *
+ *   @Override
+ *   public Pointers pointers() {
+ *     return POINTERS.view(this);
+ *   }
+ * }
+ * }</pre>
+ *
+ * @param <T> the type
+ * @since 4.17.0
+ */
+public interface PointersSupplier<T> {
+  /**
+   * Gets a new pointers supplier builder.
+   *
+   * @param <T> the type
+   * @return the builder
+   * @since 4.17.0
+   */
+  static <T> @NotNull Builder<T> builder() {
```

Comment:
actually, should this instead be `pointersSupplier`?

Reply:
im not sure, there's a lot of places we just use `builder` outside, any thoughts?

Reply:
as much as inconsistency is the spice of life, this would match our `Component` builder methods better. @kashike what do you think?

Reply:
ok, I think we're fine with this, I'm happy to just get this PR wrapped up